vllm/model_executor/layers/quantization/compressed_tensors/compressed_tensors.py

```diff
@@ -158,9 +158,36 @@ def get_quant_method(
         if isinstance(layer, Attention):
             return CompressedTensorsKVCacheMethod(self)
         if isinstance(layer, FusedMoE):
-            return CompressedTensorsMoEMethod.get_moe_method(self, layer)
+            # FusedMoE was made by combining multiple Linears so need to
+            # make sure quantization config for Linear can target it
+            self._add_fused_moe_to_target_scheme_map()
+            unfused_names = [
+                prefix + proj_name
+                for proj_name in [".0.gate_proj", ".0.up_proj", ".0.down_proj"]
+            ]
+            all_quant_schemes = set(
+                [self.get_scheme(layer, name) for name in unfused_names]
+            )
+
+            if len(all_quant_schemes) > 1:  # inconsistent qscheme
+                raise ValueError(
+                    "All MoE projections need to have same quantization scheme"
+                )
+
+            quant_scheme = all_quant_schemes.pop()
```

Comment:
I think this logic should go in `CompressedTensorsMoEMethod.get_moe_method`

Reply:
shouldn't it match the way we're doing it for linear?